flake.nix

```diff
@@ -1,50 +1,268 @@
+
+    CHaP.url = "github:input-output-hk/cardano-haskell-packages?ref=repo";
+    CHaP.flake = false;
+
+    cardano-mainnet-mirror.url = "github:input-output-hk/cardano-mainnet-mirror";
+    cardano-mainnet-mirror.flake = false;
+
+    # cicero
+    tullia.url = "github:input-output-hk/tullia";
+
+    # non-flake nix compatibility
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
+  };
+
+  outputs = inputs: let
+    supportedSystems = [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-linux"
+      "aarch64-darwin"
+    ];
+  in
+    inputs.flake-utils.lib.eachSystem supportedSystems (
+      system: let
+        # setup our nixpkgs with the haskell.nix overlays, and the iohk-nix
+        # overlays...
+        nixpkgs = import inputs.nixpkgs {
+          overlays =
+            [inputs.haskellNix.overlay]
+            ++ builtins.attrValues inputs.iohkNix.overlays
+            ++ [(import ./nix/pkgs.nix)];
+          inherit system;
+          inherit (inputs.haskellNix) config;
+        };
+        inherit (nixpkgs) lib;
+
+        # We use cabalProject' to ensure we don't build the plan for
+        # all systems.
+        cabalProject = nixpkgs.haskell-nix.cabalProject' {
+          src = ./.;
+          name = "cardano-ledger";
+          compiler-nix-name = "ghc927";
+
+          # CHaP input map, so we can find CHaP packages (needs to be more
+          # recent than the index-state we set!). Can be updated with
+          #
+          #  nix flake lock --update-input CHaP
+          #
+          inputMap = {
+            "https://input-output-hk.github.io/cardano-haskell-packages" = inputs.CHaP;
+          };
+
+          # tools we want in our shell
+          shell.tools = {
+            cabal = "3.10.1.0";
+            ghcid = "0.8.8";
+            haskell-language-server = "latest";
+            hlint = {};
+            fourmolu = "0.10.1.0";
+          };
+          # Now we use pkgsBuildBuild, to make sure that even in the cross
+          # compilation setting, we don't run into issues where we pick tools
+          # for the target.
+          shell.nativeBuildInputs = with nixpkgs.pkgsBuildBuild; [
+            (pkgs.python3.withPackages (ps: with ps; [sphinx sphinx_rtd_theme recommonmark sphinx-markdown-tables sphinxemoji]))
+            haskellPackages.implicit-hie
+          ];
+          shell.withHoogle = false;
+
+          # package customizations as needed. Where cabal.project is not
+          # specific enough, or doesn't allow setting these.
+          modules = [
+            ({pkgs, ...}: {
+              # Use our forked libsodium from iohk-nix crypto overlay.
+              packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [[pkgs.libsodium-vrf pkgs.secp256k1]];
+              packages.cardano-crypto-praos.components.library.pkgconfig = lib.mkForce [[pkgs.libsodium-vrf]];
```

Comment:
These should not be necessary anymore. iohk-nix should inject them appropriately.
```suggestion
```

Reply:
Though if we put `iohkNix.overlays.crypto` before haskell.nix overlays then https://github.com/input-output-hk/iohk-nix/blob/86421fdd89b3af43fa716ccd07638f96c6ecd1e4/overlays/crypto/default.nix#L18-L26 does not get injected right?
working on a fix.

Reply:
Ohh god. This is a bit messy 💀 loading it twice might work. But is also not ideal. Maybe this needs to be split. But in general the order dependence is rather annoying.

Reply:
fix in https://github.com/input-output-hk/iohk-nix/pull/543